diff --git a/src/balances.rs b/src/balances.rs
new file mode 100644
index 00000000..dae1c4f1
--- /dev/null
+++ b/src/balances.rs
@@ -0,0 +1 @@
+/* TODO: create a new public struct named `Pallet`. */
diff --git a/src/main.rs b/src/main.rs
index a30eb952..2b4f7670 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,3 +1,5 @@
+/* TODO: use your new module `balances` */
+
 fn main() {
 	println!("Hello, world!");
 }
