diff --git a/src/balances.rs b/src/balances.rs
index f2f602c..6a03b5b 100644
--- a/src/balances.rs
+++ b/src/balances.rs
@@ -4,7 +4,7 @@ use std::collections::BTreeMap;
 /// It is a simple module which keeps track of how much balance each account has in this state
 /// machine.
 pub struct Pallet {
-	// A simple storage mapping from accounts (`&'static str`) to their balances (`u128`).
+	// A simple storage mapping from accounts (`String`) to their balances (`u128`).
 	balances: BTreeMap<String, u128>,
 }
 
@@ -13,4 +13,17 @@ impl Pallet {
 	pub fn new() -> Self {
 		Self { balances: BTreeMap::new() }
 	}
+
+	/// Set the balance of an account `who` to some `amount`.
+	pub fn set_balance(&mut self, who: &String, amount: u128) {
+		/* Insert `amount` into the BTreeMap under `who`. */
+		unimplemented!()
+	}
+
+	/// Get the balance of an account `who`.
+	/// If the account has no stored balance, we return zero.
+	pub fn balance(&self, who: &String) -> u128 {
+		/* Return the balance of `who`, returning zero if `None`. */
+		unimplemented!()
+	}
 }
