diff --git a/src/balances.rs b/src/balances.rs
index 1b9d9d8..e09906c 100644
--- a/src/balances.rs
+++ b/src/balances.rs
@@ -3,7 +3,7 @@ use std::collections::BTreeMap;
 /// This is the Balances Module.
 /// It is a simple module which keeps track of how much balance each account has in this state
 /// machine.
-/* TODO: Add the derive macro to implement the `Debug` trait for `Pallet`. */
+#[derive(Debug)]
 pub struct Pallet {
 	// A simple storage mapping from accounts (`String`) to their balances (`u128`).
 	balances: BTreeMap<String, u128>,
diff --git a/src/main.rs b/src/main.rs
index b79655a..1ebef92 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -3,7 +3,7 @@ mod system;
 
 // This is our main Runtime.
 // It accumulates all of the different pallets we want to use.
-/* TODO: Add the derive macro to implement the `Debug` trait for `Runtime`. */
+#[derive(Debug)]
 pub struct Runtime {
 	system: system::Pallet,
 	balances: balances::Pallet,
@@ -39,5 +39,5 @@ fn main() {
 	runtime.system.inc_nonce(&alice);
 	let _res = runtime.balances.transfer(alice, charlie, 20).map_err(|e| eprintln!("{}", e));
 
-	/* TODO: Print the final runtime state after all transactions. */
+	println!("{:#?}", runtime);
 }
diff --git a/src/system.rs b/src/system.rs
index 56f3b9a..dda7af9 100644
--- a/src/system.rs
+++ b/src/system.rs
@@ -2,7 +2,7 @@ use std::collections::BTreeMap;
 
 /// This is the System Pallet.
 /// It handles low level state needed for your blockchain.
-/* TODO: Add the derive macro to implement the `Debug` trait for `Pallet`. */
+#[derive(Debug)]
 pub struct Pallet {
 	/// The current block number.
 	block_number: u32,
