diff --git a/src/main.rs b/src/main.rs
index 52343a3..8c36355 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -20,7 +20,10 @@ mod types {
 // These are all the calls which are exposed to the world.
 // Note that it is just an accumulation of the calls exposed by each module.
 pub enum RuntimeCall {
-	// TODO: Not implemented yet.
+	/* TODO: Create an enum variant `BalancesTransfer` which contains named fields:
+		- `to`: an `AccountId`
+		- `amount`: a `Balance`
+	*/
 }
 
 // This is our main Runtime.
@@ -81,7 +84,18 @@ impl crate::support::Dispatch for Runtime {
 		caller: Self::Caller,
 		runtime_call: Self::Call,
 	) -> support::DispatchResult {
-		unimplemented!();
+		/*
+			TODO:
+			Use a match statement to route the `runtime_call` to call the appropriate function in
+			our pallet. In this case, there is only `self.balances.transfer`.
+
+			Your `runtime_call` won't contain the caller information which is needed to make the
+			`transfer` call, but you have that information from the arguments to the `dispatch`
+			function.
+
+			You should propagate any errors from the call back up this function.
+		*/
+		Ok(())
 	}
 }
 
