diff --git a/src/main.rs b/src/main.rs
index d2fffb8..815fd74 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,5 +1,5 @@
 mod balances;
-/* TODO: Import your new `system` module. */
+mod system;
 
 fn main() {
 	println!("Hello, world!");
diff --git a/src/system.rs b/src/system.rs
index b4e6ef4..99d537c 100644
--- a/src/system.rs
+++ b/src/system.rs
@@ -1,17 +1,17 @@
-/* TODO: You might need to update your imports. */
+use std::collections::BTreeMap;
 
 /// This is the System Pallet.
 /// It handles low level state needed for your blockchain.
 pub struct Pallet {
 	/// The current block number.
-	/* TODO: Create a field `block_number` that stores a `u32`. */
+	block_number: u32,
 	/// A map from an account to their nonce.
-	/* TODO: Create a field `nonce` that is a `BTreeMap` from `String` to `u32`. */
+	nonce: BTreeMap<String, u32>,
 }
 
 impl Pallet {
 	/// Create a new instance of the System Pallet.
 	pub fn new() -> Self {
-		/* TODO: Return a new instance of the `Pallet` struct. */
+		Self { block_number: 0, nonce: BTreeMap::new() }
 	}
 }
