diff --git a/src/main.rs b/src/main.rs
index 597aca6..691aacc 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -10,6 +10,15 @@ mod types {
 	pub type Balance = u128;
 	pub type BlockNumber = u32;
 	pub type Nonce = u32;
+	/* TODO: Define a concrete `Extrinsic` type using `AccountId` and `RuntimeCall`. */
+	/* TODO: Define a concrete `Header` type using `BlockNumber`. */
+	/* TODO: Define a concrete `Block` type using `Header` and `Extrinsic`. */
+}
+
+// These are all the calls which are exposed to the world.
+// Note that it is just an accumulation of the calls exposed by each module.
+pub enum RuntimeCall {
+	// TODO: Not implemented yet.
 }
 
 // This is our main Runtime.
