diff --git a/src/main.rs b/src/main.rs
index f0257a6..199f7de 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -20,8 +20,7 @@ mod types {
 // These are all the calls which are exposed to the world.
 // Note that it is just an accumulation of the calls exposed by each module.
 pub enum RuntimeCall {
-	/* TODO: Turn this into a nested enum where variant `Balances` contains a `balances::Call`. */
-	BalancesTransfer { to: types::AccountId, amount: types::Balance },
+	Balances(balances::Call<Runtime>),
 }
 
 // This is our main Runtime.
@@ -85,13 +84,8 @@ impl crate::support::Dispatch for Runtime {
 		// This match statement will allow us to correctly route `RuntimeCall`s
 		// to the appropriate pallet level function.
 		match runtime_call {
-			/*
-				TODO:
-				Adjust this logic to handle the nested enums, and simply call the `dispatch` logic
-				on the balances call, rather than the function directly.
-			*/
-			RuntimeCall::BalancesTransfer { to, amount } => {
-				self.balances.transfer(caller, to, amount)?;
+			RuntimeCall::Balances(call) => {
+				self.balances.dispatch(caller, call)?;
 			},
 		}
 		Ok(())
@@ -114,14 +108,13 @@ fn main() {
 	let block_1 = types::Block {
 		header: support::Header { block_number: 1 },
 		extrinsics: vec![
-			/* TODO: Update your extrinsics to use the nested enum. */
 			support::Extrinsic {
 				caller: alice.clone(),
-				call: RuntimeCall::BalancesTransfer { to: bob, amount: 20 },
+				call: RuntimeCall::Balances(balances::Call::Transfer { to: bob, amount: 20 }),
 			},
 			support::Extrinsic {
 				caller: alice,
-				call: RuntimeCall::BalancesTransfer { to: charlie, amount: 20 },
+				call: RuntimeCall::Balances(balances::Call::Transfer { to: charlie, amount: 20 }),
 			},
 		],
 	};
