diff --git a/src/main.rs b/src/main.rs
index ca513a4..8e1547b 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -134,9 +134,17 @@ fn main() {
 		],
 	};
 
+	/*
+		TODO:
+		Create new block(s) which execute extrinsics for the new `ProofOfExistence` pallet.
+			- Make sure to set the block number correctly.
+			- Feel free to allow some extrinsics to fail, and see the errors appear.
+	*/
+
 	// Execute the extrinsics which make up our block.
 	// If there are any errors, our system panics, since we should not execute invalid blocks.
 	runtime.execute_block(block_1).expect("invalid block");
+	/* TODO: Execute your new block(s). */
 
 	// Simply print the debug format of our runtime state.
 	println!("{:#?}", runtime);
