diff --git a/src/balances.rs b/src/balances.rs
index 6a03b5b..2a6bbfb 100644
--- a/src/balances.rs
+++ b/src/balances.rs
@@ -16,14 +16,12 @@ impl Pallet {
 
 	/// Set the balance of an account `who` to some `amount`.
 	pub fn set_balance(&mut self, who: &String, amount: u128) {
-		/* Insert `amount` into the BTreeMap under `who`. */
-		unimplemented!()
+		self.balances.insert(who.clone(), amount);
 	}
 
 	/// Get the balance of an account `who`.
 	/// If the account has no stored balance, we return zero.
 	pub fn balance(&self, who: &String) -> u128 {
-		/* Return the balance of `who`, returning zero if `None`. */
-		unimplemented!()
+		*self.balances.get(who).unwrap_or(&0)
 	}
 }
