diff --git a/src/main.rs b/src/main.rs
index 95b5760..2d9887b 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -4,15 +4,14 @@ mod system;
 // This is our main Runtime.
 // It accumulates all of the different pallets we want to use.
 pub struct Runtime {
-	/* TODO: create a field `system` which is of type `system::Pallet`. */
-	/* TODO: create a field `balances` which is of type `balances::Pallet`. */
+	system: system::Pallet,
+	balances: balances::Pallet,
 }
 
 impl Runtime {
 	// Create a new instance of the main Runtime, by creating a new instance of each pallet.
 	fn new() -> Self {
-		/* TODO: Create a new `Runtime` by creating new instances of `system` and `balances`. */
-		unimplemented!()
+		Self { system: system::Pallet::new(), balances: balances::Pallet::new() }
 	}
 }
 
